core/src/HIP/Kokkos_HIP_Parallel_Range.hpp

```diff
@@ -113,10 +115,246 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
         false);
   }
 
-  ParallelFor(const FunctorType &arg_functor, const Policy &arg_policy)
+  ParallelFor(const FunctorType& arg_functor, const Policy& arg_policy)
       : m_functor(arg_functor), m_policy(arg_policy) {}
 };
 
+//----------------------------------------------------------------------------
+//----------------------------------------------------------------------------
+
+template <class FunctorType, class ReducerType, class... Traits>
+class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
+                     Kokkos::Experimental::HIP> {
+ public:
+  using Policy = Kokkos::RangePolicy<Traits...>;
+
+ private:
+  using WorkRange    = typename Policy::WorkRange;
+  using WorkTag      = typename Policy::work_tag;
+  using Member       = typename Policy::member_type;
+  using LaunchBounds = typename Policy::launch_bounds;
+
+  using ReducerConditional =
+      Kokkos::Impl::if_c<std::is_same<InvalidType, ReducerType>::value,
+                         FunctorType, ReducerType>;
+  using ReducerTypeFwd = typename ReducerConditional::type;
+  using WorkTagFwd =
+      typename Kokkos::Impl::if_c<std::is_same<InvalidType, ReducerType>::value,
+                                  WorkTag, void>::type;
+
+  using ValueTraits =
+      Kokkos::Impl::FunctorValueTraits<ReducerTypeFwd, WorkTagFwd>;
+  using ValueInit = Kokkos::Impl::FunctorValueInit<ReducerTypeFwd, WorkTagFwd>;
+  using ValueJoin = Kokkos::Impl::FunctorValueJoin<ReducerTypeFwd, WorkTagFwd>;
+
+ public:
+  using pointer_type   = typename ValueTraits::pointer_type;
+  using value_type     = typename ValueTraits::value_type;
+  using reference_type = typename ValueTraits::reference_type;
+  using functor_type   = FunctorType;
+  using size_type      = Kokkos::Experimental::HIP::size_type;
+  using index_type     = typename Policy::index_type;
+
+  // Algorithmic constraints: blockSize is a power of two AND hipBlockDim_y ==
+  // hipBlockDim_z == 1
+
+  const FunctorType m_functor;
+  const Policy m_policy;
+  const ReducerType m_reducer;
+  const pointer_type m_result_ptr;
+  const bool m_result_ptr_device_accessible;
+  size_type* m_scratch_space;
+  size_type* m_scratch_flags;
+
+  // Shall we use the shfl based reduction or not (only use it for static sized
+  // types of more than 128bit)
+  enum {
+    UseShflReduction = false
+  };  //((sizeof(value_type)>2*sizeof(double)) && ValueTraits::StaticValueSize)
+      //};
```

Comment:
The commented expression was replaced by `false`?

Reply:
Yes, but actually in code is deactivated in the same way in CUDA. So the shuffle reduction is never used but the code is there.